core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt

```diff
@@ -514,6 +514,16 @@ object NextTurnAutomation {
     }
 
     private fun chooseTechToResearch(civInfo: Civilization) {
+        while(civInfo.tech.freeTechs > 0) {
+            val researchableTechs = civInfo.gameInfo.ruleset.technologies.values
```

Comment:
I changed the .toSortedMap to a list of lists since costs[0] causes an error because the key is the actual research value. The previous code used techsGroups[costs[1]]!!, which is the same as using regular integer indexing but needs a list of keys as well.